java/fury-core/src/main/java/io/fury/Fury.java

```diff
@@ -707,7 +707,14 @@ public Object deserialize(InputStream inputStream, Iterable<MemoryBuffer> outOfB
       Preconditions.checkArgument(read == 4);
       int size = buffer.readInt();
       buffer.ensure(size + 4);
-      read = inputStream.read(buffer.getHeapMemory(), 4, size);
+      read = 0;
```

Comment:
Do we need to extract this part as a private method? It's repeated twice.

Reply:
I will refactor and add a unit test.

Reply:
Let me know if you want any more changes or feel free to update the code so that it fits your personal style.